### PR TITLE
[MAHOUT-1853] Implementing first part

### PR DESCRIPTION
**Pull request by pferrel**

### CODE REVIEW

File: math-scala/src/main/scala/org/apache/mahout/math/cf/SimilarityAnalysis.scala [hunk 1]
```diff
@@ -211,9 +314,17 @@ object SimilarityAnalysis extends Serializable {
...
+  def computeSimilarities(
+    drm: DrmLike[Int],
+    numUsers: Int,
+    maxInterestingItemsPerThing: Int,
+    bcastNumInteractionsB: BCast[Vector],
+    bcastNumInteractionsA: BCast[Vector],
+    crossCooccurrence: Boolean = true,
+    minLLROpt: Option[Double] = None) = {
+
+    val minLLR = minLLROpt.getOrElse(0.0d) // accept all values if not specified
```
Comment (dlyubimov) [inline math-scala/src/main/scala/org/apache/mahout/math/cf/SimilarityAnalysis.scala]: i think style convention was to use 0.0 (minority split in favor 0d) but never 0.0d

File: spark/src/test/scala/org/apache/mahout/cf/SimilarityAnalysisSuite.scala [hunk 2]
```diff
@@ -91,13 +93,13 @@ class SimilarityAnalysisSuite extends FunSuite with MahoutSuite with Distributed
...
     n = (new MatrixOps(m = diff2Matrix)).norm
     n should be < 1E-10
 
   }
 
-  test("cooccurrence [A'A], [B'A] double data using LLR") {
+  test("Cross-occurrence [A'A], [B'A] double data using LLR") {
     val a = dense(
         (100000.0D, 1.0D,  0.0D,  0.0D,     0.0D),
         (     0.0D, 0.0D, 10.0D,  1.0D,     0.0D),
```
Comment (dlyubimov) [inline spark/src/test/scala/org/apache/mahout/cf/SimilarityAnalysisSuite.scala]: same note. either 0.0 or 0d but not 0.0d

File: spark/src/test/scala/org/apache/mahout/cf/SimilarityAnalysisSuite.scala [hunk 3]
```diff
@@ -191,14 +193,115 @@ class SimilarityAnalysisSuite extends FunSuite with MahoutSuite with Distributed
...
+    val users = Seq("u1", "u2", "u3", "u4")
+    val itemsA = Seq("a1", "a2", "a3", "a4", "a5")
+    val itemsB = Seq("b1", "b2", "b3", "b4")
+    val userDict = new BiDictionary(users)
+    val itemsADict = new BiDictionary(itemsA)
+    val itemsBDict = new BiDictionary(itemsB)
+
+    // this is downsampled to the top 2 values per row to match the calc
+    val matrixLLRCoocAtBNonSymmetric = dense(
+      (0.0,                1.7260924347106847, 1.7260924347106847, 0.0),
```
Comment (dlyubimov) [inline spark/src/test/scala/org/apache/mahout/cf/SimilarityAnalysisSuite.scala]: this is our accepted convention, good

**Comment by andrewpalumbo** [comment]
@pferrel is this something that needs to go into 0.13.0?

**Comment by pferrel** [comment]
It has already been merged. The style comments are valid but not required. Just saw them now (2/26/2017)

**Pull request closed without merging**